pkg/utils/revision/revision_utils.go

```diff
@@ -0,0 +1,301 @@
+package revision
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"hash"
+	"hash/fnv"
+
+	"github.com/davecgh/go-spew/spew"
+	appsv1 "k8s.io/api/apps/v1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	leaderworkerset "sigs.k8s.io/lws/api/leaderworkerset/v1"
+)
+
+// Functions in this package are adapted from https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/statefulset/ and
```

Comment:
I would create a GetRevisionKey func as follows and use it instead of access the label directly in many places; this also allows us to check for nil label in one place:

```
func GetRevisionKey(obj metav1.Object) string {
  if obj.Labels != nil {
    return obj.Labels[leaderworkerset.TemplateRevisionHashKey]
  }
  return ""
}
```

Reply:
this function can be used by pod, sts and revision objects